gen2-efficientDet/main.py

```diff
@@ -0,0 +1,88 @@
+from pathlib import Path
+import numpy as np
+import cv2
+import depthai as dai
+import time
+
+CONF_THRESHOLD = 0.4
+SHAPE = 320
+coco_90 = ["person", "bicycle", "car", "motorcycle", "airplane", "bus", "train", "truck", "boat", "traffic light", "fire hydrant", "12", "stop sign", "parking meter", "bench", "bird", "cat", "dog", "horse", "sheep", "cow", "elephant", "bear", "zebra", "giraffe", "26", "backpack", "umbrella", "29", "30", "handbag", "tie", "suitcase", "frisbee", "skis", "snowboard", "sports ball", "kite", "baseball bat", "baseball glove", "skateboard", "surfboard", "tennis racket", "bottle", "45", "wine glass", "cup", "fork", "knife", "spoon", "bowl", "banana", "apple", "sandwich", "orange", "broccoli", "carrot", "hot dog", "pizza", "donut", "cake", "chair", "couch", "potted plant", "bed", "66", "dining table", "68", "69", "toilet", "71", "tv", "laptop", "mouse", "remote", "keyboard", "cell phone", "microwave", "oven", "toaster", "sink", "refrigerator", "83", "book", "clock", "vase", "scissors", "teddy bear", "hair drier", "toothbrush"]
+
+p = dai.Pipeline()
+p.setOpenVINOVersion(dai.OpenVINO.VERSION_2021_3)
+
+class FPSHandler:
+    def __init__(self, cap=None):
+        self.timestamp = time.time()
+        self.start = time.time()
+        self.frame_cnt = 0
+    def next_iter(self):
+        self.timestamp = time.time()
+        self.frame_cnt += 1
+    def fps(self):
+        return self.frame_cnt / (self.timestamp - self.start)
+
+camRgb = p.createColorCamera()
+camRgb.setPreviewSize(SHAPE, SHAPE)
+camRgb.setResolution(dai.ColorCameraProperties.SensorResolution.THE_1080_P)
+camRgb.setInterleaved(False)
+camRgb.setColorOrder(dai.ColorCameraProperties.ColorOrder.BGR)
+camRgb.setFp16(True) # Model requires FP16 input
+
+# NN that detects faces in the image
+nn = p.createNeuralNetwork()
+nn.setBlobPath(str(Path("models/efficientdet_lite0_2021.3_6shaves.blob").resolve().absolute()))
+nn.setNumInferenceThreads(2)
+camRgb.preview.link(nn.input)
+
+# Send bouding box from the NN to the host via XLink
+nn_xout = p.createXLinkOut()
+nn_xout.setStreamName("nn")
+nn.out.link(nn_xout.input)
+# Send rgb frames to the host
+rgb_xout = p.createXLinkOut()
+rgb_xout.setStreamName("rgb")
+nn.passthrough.link(rgb_xout.input)
+
+# Pipeline is defined, now we can connect to the device
+with dai.Device(p) as device:
+    device.startPipeline()
+    qRgb = device.getOutputQueue(name="rgb", maxSize=4, blocking=False)
+    qNn = device.getOutputQueue(name="nn", maxSize=4, blocking=False)
+    fps = FPSHandler()
+    shape = (3, SHAPE, SHAPE)
+
+    while True:
+        inRgb = qRgb.get()
+        data = inRgb.getData()
+        # Model needs FP16 so we have to convert color frame back to U8 on the host
+        frame = np.array(data).view(np.float16).reshape(shape).transpose(1, 2, 0).astype(np.uint8).copy()

```

Comment:
it can be simplified to `np.array(inRgb.getData())`, we will avoid creating `data` variable which is later overwritten and is not used in this form